internal/hardware/validator.go

```diff
@@ -48,7 +50,7 @@ func (v *validator) GetHostValidDisks(host *models.Host) ([]*models.Disk, error)
 	if err := json.Unmarshal([]byte(host.Inventory), &inventory); err != nil {
 		return nil, err
 	}
-	disks := ListValidDisks(&inventory, gbToBytes(v.MinDiskSizeGb))
```

Comment:
The burden of checking which disks are valid has been shifted from ListValidDisks to UpdateInventory. Now ListValidDisks simply parses the new eligibility struct in order to "list" (and sort) the valid disks.

That's why it no longer needs the MinDiskSize parameter.